internal/handlers/pg/pgdb/query.go

```diff
@@ -299,11 +300,11 @@ func prepareWhereClause(p *Placeholder, sqlFilters *types.Document) (string, []a
 						// type not supported for pushdown
 
 					case float64, bool, int32, int64:
-						filters = append(filters, fmt.Sprintf(sql, p.Next(), p.Next(), sjson.GetTypeOfValue(v)))
+						filters = append(filters, fmt.Sprintf(sql, p.Next(), p.Next(), commonparams.AliasFromType(v)))
 						args = append(args, rootKey, v)
 
 					case string, types.ObjectID, time.Time:
-						filters = append(filters, fmt.Sprintf(sql, p.Next(), p.Next(), sjson.GetTypeOfValue(v)))
+						filters = append(filters, fmt.Sprintf(sql, p.Next(), p.Next(), commonparams.AliasFromType(v)))
```

Comment:
Our pg supports less types than bson does (`commonparams.AliasFromType`), so for this usage we want stricter `sjson.GetTypeOfValue` I think.